packages/editor/src/components/rich-text/index.js

```diff
@@ -780,6 +778,11 @@ export class RichText extends Component {
 			record.end = length;
 			this.applyRecord( record );
 		}
+
+		if ( propsToCheck.some( ( name ) => this.props[ name ] !== prevProps[ name ] ) ) {
```

Comment:
Maybe needs better name. Also there needs to be mechanism in place to prevent prop name clashes...